### ISSUE TITLE
Have workflows only run when files that they check are changed

### ISSUE THREAD

**Issue by andrewtavis** [scribe-org/Scribe-Data#506]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-Data/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-Data's [Code of Conduct](https://github.com/scribe-org/Scribe-Data/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

As of now we're always running the project and query check workflows as found in [src/scribe_data/check](https://github.com/scribe-org/Scribe-Data/tree/main/src/scribe_data/check), but ideally they would only be ran under certain conditions:

- check_project_metadata.py: When metadata files have been changed
- check_project_structure.py: Likely can stay as is
- check_pyicu.py: Likely can stay as is
- check_query_forms.py: When SPARQL queries are edited
- check_query_identifiers.py: When SPARQL queries are edited

The reason to make these changes is to assure that we're not running workflows that aren't needed for GitHub Actions allotment and environmental reasons :)

### Contribution

Happy to review and work on this with someone! 😊

**Comment by andrewtavis** [comment]
CC @DeleMike and @catreedle 👋 Would either/both of you have interest in looking into this?

**Comment by DeleMike** [comment]
@andrewtavis I would love to work on this issue. Thank you!

**Comment by andrewtavis** [comment]
Thank you, @DeleMike! Let us know if any questions come up. Maybe you can do it for `check_query_forms.py` and `check_query_identifiers.py`, and then @catreedle can work on it for `check_project_metadata.py`?

Let me know as well if you agree that it'd be hard to have a trigger for actions for `check_project_structure .py` and that we don't need one for `check_pyicu.py` :)

**Comment by DeleMike** [comment]
Sure! Once I have questions, I will ask :)

**Comment by DeleMike** [comment]
Sorry @andrewtavis, I had a few things come up. I'm looking into this now.

**Comment by andrewtavis** [comment]
No stress at all, @DeleMike! Looking forward to the solutions 😊

**Comment by DeleMike** [comment]
Thanks @andrewtavis , one issue I have is pulling the updated codebase. I keep having this error:
```bash
mac@MACs-MacBook-Pro ~/S/S/s/s/w/language_data_extraction (main)> git pull upstream main               
                   
remote: Enumerating objects: 775, done.
remote: Counting objects: 100% (775/775), done.
remote: Compressing objects: 100% (254/254), done.
error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly: CANCEL (err 8)
error: 7476 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
``` 

I have been trying for so long to fix this.

**Comment by andrewtavis** [comment]
Hey @DeleMike 👋 I guess I'm a bit confused by your directory structure 🤔 Your local Scribe-Data is `language_data_extraction`? Worst comes to worse you could also delete your local copy and redo the setup steps in the readme with the most recent version of main?

Btw: Seeing you're on Mac, would you have interest in working on the iOS app at some point?

**Comment by andrewtavis** [comment]
Maybe the best thing to since it's been so long is to just recreate the project, @DeleMike?

> Yes, I would love too! I will need to learn iOS Dev. I would finally learn iOS Dev...lol

Nice! Let's look into an iOS issue for you at some point then :)

**Comment by DeleMike** [comment]
@andrewtavis yes I have tried to reclone the repo and it is the same issue.

<img width="675" alt="Screenshot 2024-11-17 at 17 06 44" src="https://github.com/user-attachments/assets/44a67229-e83b-4730-afd2-78e05d2a5556">

If you see the screenshot, you can see where my issue lies. I was even able to clone another repo

**Comment by DeleMike** [comment]
never mind! omg! It has successfully pulled!!! 😭

**Comment by andrewtavis** [comment]
Nice, glad it's working, @DeleMike! 🚀

**Comment by andrewtavis** [comment]
Closed by #508 🚀 Thanks for the continued support for the workflows, @DeleMike!
